backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/predicate/impl/ElasticsearchSearchPredicate.java

```diff
@@ -24,8 +25,14 @@ public interface ElasticsearchSearchPredicate extends SearchPredicate {
 
 	void checkNestableWithin(String expectedParentNestedPath);
 
+	default ElasticsearchSearchPredicate checkAcceptableAsBoolPredicateClause(String clauseType) {
+		return this;
+	}
```

Comment:
This is dangerous, as composite predicates (i.e. `bool`, `nested`) need to propagate the check to their children.

I'd recommend not having a default implementation here, or moving the default to a more specific class that can be reasonably assumed to be usable in any bool clause, e.g. `AbstractElasticsearchSingleFieldPredicate`.